Serverless restore timeout (#20920)
pingcap/docs: tidb-cloud/backup-and-restore-serverless.md

- User credentials and permissions from the source cluster will not be restored to the new cluster.

> ### Restore Timeout

Reviewer: ```suggestion
### Restore timeout
```

> The restore process typically completes within a few minutes. However, it will be automatically canceled when restore exceeds 3 hours. The outcome of a canceled restore depends on the destination:

Reviewer: ```suggestion
The restore process typically completes within a few minutes. If the restore takes longer than three hours, it is automatically canceled. The outcome of a canceled restore depends on the destination:
```

> - **Restore to a New Cluster**: The target cluster will be deleted and the source cluster will remain unchanged.
> - **Restore In-Place**: The cluster status will be changed to **Available** from **Restoring**, and it will become accessible again. 

Reviewer: ```suggestion
- **In-place restore**: the cluster status changes from **Restoring** to **Available**, and the cluster becomes accessible.
- **Restore to a new cluster**: the new cluster is deleted and the source cluster remains unchanged.
```
Keep the order consistent with the "Restore destination" section
https://github.com/pingcap/docs/blob/8235d963f467320197177f2ec15aa12055528808/tidb-cloud/backup-and-restore-serverless.md?plain=1#L65-L77
